src/engine/source/builder/builders/opBuilderKVDB.cpp

```diff
@@ -446,4 +446,363 @@ HelperBuilder getOpBuilderKVDBDelete(std::shared_ptr<IKVDBManager> kvdbManager,
+        checkParametersSize(rawName, parameters, 2);
+        const auto name = formatHelperName(targetField, rawName, parameters);
+
+        const auto& dbName = parameters[0].m_value;
+
+        if (Parameter::Type::VALUE != parameters[0].m_type)
+        {
+            throw std::runtime_error(fmt::format("Engine KVDB builder: {}.", "DB Name must be a value"));
+        }
+
+        if (Parameter::Type::REFERENCE != parameters[1].m_type)
+        {
+            throw std::runtime_error(fmt::format("Engine KVDB builder: {}.", "Key array must be a reference"));
+        }
+
+        auto arrayRef = parameters[1].m_value;
+
+        // Check target field is an array
+        if (schema->hasField(targetField) && schema->getType(targetField) != json::Json::Type::Array)
+        {
+            throw std::runtime_error(fmt::format("Engine KVDB builder: {}.", "Target field must be an array"));
+        }
+
+        // Check array reference is an array
+        if (schema->hasField(arrayRef) && schema->getType(arrayRef) != json::Json::Type::Array)
+        {
+            throw std::runtime_error(fmt::format("Engine KVDB builder: {}.", "Array reference must be an array"));
+        }
+
+        // Get KVDB handler
+        auto resultHandler = kvdbManager->getKVDBHandler(dbName, kvdbScopeName);
+        if (std::holds_alternative<base::Error>(resultHandler))
+        {
+            throw std::runtime_error(
+                fmt::format("Engine KVDB builder: {}.", std::get<base::Error>(resultHandler).message));
+        }
+
+        // Trace messages
+        auto referenceNotFound = fmt::format("Reference '{}' not found or not array", arrayRef);
+        auto emptyArray = fmt::format("Empty array in reference '{}'", arrayRef);
+        auto notKeyStr = fmt::format("Found non-string key in reference '{}'", arrayRef);
+        auto heterogeneousTypes = fmt::format("Heterogeneous types when obtaining values from database '{}'", dbName);
+        auto malformedJson = fmt::format("Malformed JSON found in database '{}'", dbName);
+        auto keyNotMatched = fmt::format("Key not found in DB '{}'", dbName);
+
+        auto successTrace = fmt::format("[{}] -> Success", name);
+
+        // Return Expression
+        return base::Term<base::EngineOp>::create(
+            name,
+            [=, kvdbHandler = std::get<std::shared_ptr<kvdbManager::IKVDBHandler>>(resultHandler)](base::Event event)
+            {
+                // Resolve array of keys reference
+                auto keys = event->getArray(arrayRef);
+                if (!keys)
+                {
+                    return base::result::makeFailure(event, referenceNotFound);
+                }
+                if (keys.value().empty())
+                {
+                    return base::result::makeFailure(event, emptyArray);
+                }
+
+                // Get values from KVDB
+                bool first = true;
```

Comment:
improve-desing: ditto. I think we can use just one variable

Reply:
same